libpython27.dll.a is required for sip4

Apparently I modified this on my system and I had forgotten, so building sip4 currently fails, sorry :-/

Here is the build error from sip4:

```
$ makepkg-mingw -L --nocheck
=> WARNING: You don't have installed mingw-w64 toolchain for architecture x86_64.
=> WARNING: To install it run: 'pacman -S mingw-w64-x86_64-toolchain'
==> Making package: mingw-w64-i686-python2-sip4 4.16.2-1 (Thu, Jul 24, 2014  5:25:03 PM)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found sip-4.16.2.zip
==> Validating source files with md5sums...
    sip-4.16.2.zip ... Passed
==> Extracting sources...
  -> Extracting sip-4.16.2.zip with /usr/bin/bsdtar
==> Starting prepare()...
==> Removing existing $pkgdir/ directory...
==> Starting build()...
This is SIP 4.16.2 for Python 2.7.8 on win32.
The SIP code generator will be installed in
C:/Users/saghul/work/MINGW-packages/mingw-w64-python2-sip4/pkg/mingw-w64-i686-python2-sip4/mingw32/bin.
The sip module will be installed in
C:/Users/saghul/work/MINGW-packages/mingw-w64-python2-sip4/pkg/mingw-w64-i686-python2-sip4/mingw32/lib/python2.7/site-packages.
The sip.h header file will be installed in
C:/Users/saghul/work/MINGW-packages/mingw-w64-python2-sip4/pkg/mingw-w64-i686-python2-sip4/mingw32/include/python2.7.
The default directory to install .sip files in is
C:/Users/saghul/work/MINGW-packages/mingw-w64-python2-sip4/pkg/mingw-w64-i686-python2-sip4/mingw32/sip.
Creating siplib/sip.h...
Creating siplib/siplib.c...
Creating siplib/siplib.sbf...
Creating sipconfig.py...
Creating top level Makefile...
Creating sip code generator Makefile...
Creating sip module Makefile...
make[1]: Entering directory '/c/Users/saghul/work/MINGW-packages/mingw-w64-python2-sip4/src/sip-4.16.2/sipgen'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/c/Users/saghul/work/MINGW-packages/mingw-w64-python2-sip4/src/sip-4.16.2/sipgen'
make[1]: Entering directory '/c/Users/saghul/work/MINGW-packages/mingw-w64-python2-sip4/src/sip-4.16.2/siplib'
gcc -c -O2 -w -DNDEBUG -DUNICODE -DQT_LARGEFILE_SUPPORT -I. -IC:\msys2\mingw32\include\python2.7 -o siplib.o siplib.c
gcc -c -O2 -w -DNDEBUG -DUNICODE -DQT_LARGEFILE_SUPPORT -I. -IC:\msys2\mingw32\include\python2.7 -o apiversions.o apiversions.c
gcc -c -O2 -w -DNDEBUG -DUNICODE -DQT_LARGEFILE_SUPPORT -I. -IC:\msys2\mingw32\include\python2.7 -o descriptors.o descriptors.c
gcc -c -O2 -w -DNDEBUG -DUNICODE -DQT_LARGEFILE_SUPPORT -I. -IC:\msys2\mingw32\include\python2.7 -o qtlib.o qtlib.c
gcc -c -O2 -w -DNDEBUG -DUNICODE -DQT_LARGEFILE_SUPPORT -I. -IC:\msys2\mingw32\include\python2.7 -o threads.o threads.c
gcc -c -O2 -w -DNDEBUG -DUNICODE -DQT_LARGEFILE_SUPPORT -I. -IC:\msys2\mingw32\include\python2.7 -o objmap.o objmap.c
gcc -c -O2 -w -DNDEBUG -DUNICODE -DQT_LARGEFILE_SUPPORT -I. -IC:\msys2\mingw32\include\python2.7 -o voidptr.o voidptr.c
gcc -c -O2 -w -DNDEBUG -DUNICODE -DQT_LARGEFILE_SUPPORT -I. -IC:\msys2\mingw32\include\python2.7 -o array.o array.c
g++ -c -O2 -w -DNDEBUG -DUNICODE -DQT_LARGEFILE_SUPPORT -I. -IC:\msys2\mingw32\include\python2.7 -o bool.o bool.cpp
g++ -mthreads -Wl,-enable-stdcall-fixup -Wl,-enable-auto-import -Wl,-enable-runtime-pseudo-reloc -shared -Wl,-subsystem,console -Wl,-s -o sip.pyd siplib.o apiversions.o descriptors.o qtlib.o threads.o objmap.o voidptr.o array.o bool.o -LC:\msys2\mingw32\libs -lpython27
C:/msys2/mingw32/bin/../lib/gcc/i686-w64-mingw32/4.9.1/../../../../i686-w64-mingw32/bin/ld.exe: cannot find -lpython27
collect2.exe: error: ld returned 1 exit status
Makefile:36: recipe for target 'sip.pyd' failed
make[1]: *** [sip.pyd] Error 1
make[1]: Leaving directory '/c/Users/saghul/work/MINGW-packages/mingw-w64-python2-sip4/src/sip-4.16.2/siplib'
Makefile:3: recipe for target 'all' failed
make: *** [all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
```

For some reason it cannot find -lpython27. I solved it like this:

```
 cp /mingw32/lib/libpython2.7.dll.a /mingw32/lib/libpython27.dll.a
```

Is this something that can be solved in the sip4 package or should we copy the file when installing, in the python2 package? Thanks!

--- a/mingw-w64-python2-sip4/0001-fix-libpython-name.patch
+++ b/mingw-w64-python2-sip4/0001-fix-libpython-name.patch
@@ -1,0 +1,11 @@
+--- siputils.py.orig	2014-07-24 18:27:50.609200000 +0200
++++ siputils.py	2014-07-24 18:28:26.021200000 +0200
+@@ -448,7 +448,7 @@
+             elif sys.platform == "win32":
+                 libdir.append(self.config.py_lib_dir)
+ 
+-                py_lib = "python%u%u" % ((self.config.py_version >> 16), ((self.config.py_version >> 8) & 0xff))
++                py_lib = "python%u.%u" % ((self.config.py_version >> 16), ((self.config.py_version >> 8) & 0xff))
+ 
+                 # For Borland use the OMF version of the Python library if it
+                 # exists, otherwise assume that Python was built with Borland

--- a/mingw-w64-python2-sip4/PKGBUILD
+++ b/mingw-w64-python2-sip4/PKGBUILD
@@ -4,7 +4,7 @@ _realname=python2-sip4
 
 pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
 pkgver=4.16.2
-pkgrel=1
+pkgrel=2
 pkgdesc="Tool to create Python bindings for C and C++ libraries"
 arch=('any')
 license=('GPL')
@@ -12,11 +12,16 @@ url="http://www.riverbankcomputing.co.uk/software/sip"
 depends=("${MINGW_PACKAGE_PREFIX}-python2"
          "${MINGW_PACKAGE_PREFIX}-qt4")
 
-source=("http://downloads.sourceforge.net/project/pyqt/sip/sip-$pkgver/sip-$pkgver.zip")
-md5sums=('8e651c3216d247444967189e5553e750')
+source=("http://downloads.sourceforge.net/project/pyqt/sip/sip-$pkgver/sip-$pkgver.zip"
+        "0001-fix-libpython-name.patch")
+md5sums=('8e651c3216d247444967189e5553e750'
+         '114b8eb3be19ed90fded76531454eb3a')
 
 prepare() {
   cd "${srcdir}"/sip-$pkgver
+
+  # Apply patches
+  patch -p0 -i "${srcdir}"/0001-fix-libpython-name.patch
 }
 
 build() {

